### ISSUE TITLE
The function 'cost_normalization' do not use the backend and only use numpy

### ISSUE THREAD

**Issue by framunoz** [PythonOT/POT#465]
## Describe the bug
I was using the `ot.da.SinkhornTransport` class when I got the following warning:
```UserWarning: Warning: numerical errors at iteration 0```.

Thinking about why I got this error, I remembered that cost matrices should be normalised, so I changed the `norm` parameter to `"max"` to normalise it to the maximum value. However, this gave me an error.

The reason was that the `utils.cost_normalization` function is implemented only using numpy, when I'm currently using pytorch tensors. Internally, I got the backend of the `C` array that uses this function, and changed the line `C /= float(np.max(C))` to `C /= float(nx.max(C))`, and then the program worked fine.

I mention this mainly because it may be a bug that happens to someone else in another context.

### Environment (please complete the following information):
- OS (e.g. MacOS, Windows, Linux): Window
- Python version: 3.10
- How was POT installed (source, `pip`, `conda`): pip
- Build command you used (if compiling from source):
- Only for GPU related bugs:
    - CUDA version:
    - GPU models and configuration:
    - Any other relevant information:

Output of the following code snippet:
```python
import platform; print(platform.platform())
import sys; print("Python", sys.version)
import numpy; print("NumPy", numpy.__version__)
import scipy; print("SciPy", scipy.__version__)
import ot; print("POT", ot.__version__)
```

```
Windows-10-10.0.19045-SP0
Python 3.10.4 (tags/v3.10.4:9d38120, Mar 23 2022, 23:13:41) [MSC v.1929 64 bit (AMD64)]
NumPy 1.24.1
SciPy 1.9.3
POT 0.8.2
```

**Comment by rflamary** [comment]
Good point and your fix seems perfect, those are the edge cases that were not detected when converting to backend.

Could you do this short PR for us please (and add a test that check that normalization does not breaks with non-numpy arrays)?

**Comment by framunoz** [comment]
Hi,

I was writing the code to try to fix this bug in the branch "`cost_normalization_debug`". In order to fix this bug, I had to create the `median` method in the backend class, and implement this method in the rest of the backends. However, I realised that tensorflow doesn't have a `median` function, although it may be replicable with the percentile function (as seen in this Stack Overflow question [Tensorflow median value](https://stackoverflow.com/questions/43824665/tensorflow-median-value)). Also, I couldn't run the test scripts, as I didn't get the Jax library installed correctly.

Meanwhile, I will upload the above-mentioned branch so that you can see the changes made.

**Comment by rflamary** [comment]
hello, thanks @framunoz , 

you ca,n open a pull request and we can run all texts on gitrhub action on all backends (jax included). Note that It is also OK to test for simple men/max normalization and to rais a not implemented error for median normalization if tensorflow backend (which would be a simpler PR)

**Comment by framunoz** [comment]
Hello,

I tried to do the PR, but I got the following message:

> Can't create a new pull request: Push failed: remote: Permission to PythonOT/POT.git denied to framunoz. unable to access 'https://github.com/PythonOT/POT.git/': The requested URL returned error: 403
